xarray/core/common.py

```diff
@@ -70,14 +70,51 @@ def wrapped_func(self, dim=None, keep_attrs=False, **kwargs):
 
 
 class ImplementsRollingArrayReduce(object):
```

Comment:
I recognize that you didn't write this class, but I wonder if we couldn't simply consolidate it into `DataArrayRolling`? It feels like one more level of complexity than is necessary. The `ImplementsArrayReduce` type classes were implemented because we have multiple array types for which we wanted similar reduce functionality.

Either way, it would be nice to move this logic into the `rolling.py` module, just to keep it in fewer places.

Perhaps @jhamman recalls why we have this separate class...